### ISSUE TITLE
Cannot parse some solc 0.6.x language features.

### ISSUE THREAD

**Issue by cgewecke** [sc-forks/solidity-coverage#464]
Contracts which use new syntax crash the parser:
+ try/catch
+ override
+ array slicing
+ assembly `leave`
+ abstract keyword

Fixing this will require updates to the [antlr grammar][1] and [solidity-parser-antlr][2].

Example error reported in gitter by cyril lapinte
```
interface/IERC20.sol
solidity-coverage cleaning up, shutting down ganache server
ParserError: Could not instrument: interface/IERC20.sol. 
(Please verify solc can compile this file without errors.) 
extraneous input 'abstract' expecting {<EOF>, 'pragma', 'import', 'contract', 'interface', 'library'} (9:0)
```
[1]: https://github.com/solidityj/solidity-antlr4
[2]: https://github.com/federicobond/solidity-parser-antlr

**Comment by nventuro** [comment]
From [this comment](https://github.com/federicobond/solidity-parser-antlr/issues/98#issuecomment-573012235), it looks like Consensys Dillegence took up maintenance of the antlr parser [in their own fork](https://github.com/consensys/solidity-parser-antlr). 

It'd be great for coverage to upgrade and support 0.6, since migrating to this version is a complex process that could easily introduce errors (see the [PR for the OpenZeppelin Contracts initial migration](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2080)).

**Comment by cgewecke** [comment]
@nventuro 

Ok, yes - will do this now and publish over the weekend. Was waiting a bit to see how things unfolded in parser-antlr but it looks like the diligence fork is the simplest way forward atm. 

Is there a Zeppelin branch where you are no longer using the rotcivegaf fork of solidity-coverage?

**Comment by nventuro** [comment]
No, in the end we never got around to migrating to v7, sorry about that. The `dev-v3.0` branch contains the 0.6 migration, and while it does use [Test Environment](https://docs.openzeppelin.com/test-environment), it does so in a very straightforward manner, using mocha with a single underlying ganache-core instance. 

I suspect it wouldn't be too hard to support coverage in that scenario, the issues we found come up when we try to generalize the solution for any Test Environment setup, including parallel tests.

**Comment by cgewecke** [comment]
@nventuro 

Oh no worries - I will just run the new parser over the contracts in `dev-v3.0` to make sure it doesn't crash. Otherwise everything should be ok.

**Comment by cgewecke** [comment]
Cross-ref note:

Looks like diligence resolved all the syntax issues listed in the initial comment. There's a problem parsing `payable(<address>)`.

PR to resolve at [consensys/solidity-antlr4 3][1]

[1]: https://github.com/ConsenSys/solidity-antlr4/pull/3

**Comment by cgewecke** [comment]
With the caveat about `payable(<address>)` , this should be mostly fixed with `0.7.2`
